blog/_posts/2021-08-20-symbiosis-symbulation.md

```diff
@@ -0,0 +1,83 @@
+
+  While symbiosis has been simulated using Symbulation, the purpose of this project is to create and understand various types of complex symbiotic relationships while specificallly using the Avida complex genome to make the host and its symbionts.
+
+  For this summer, we focused on being able to add symbionts to a host system which would then increase the count of a point system. The original idea is that if the host reaches a certain amount of points, it would then donate a point to the symbiont and if the symbiont reaches a certain amount of points, then the symbiont would donate a point to the host which would then be multiplied by a synergy factor. The synergy factor represents the symbiont being able to provide resources to the host that it couldn't by itself, like the clownfish providing nutrients to the anemone that it wouldn't be able to get on its own.
+
+---
+
+ ## Symbiosis Environment
+ In order to create symbiosis, we need to create a host with symbionts and create a world to put them in. Header files for the world, host and symbiont were created in order to do so. These files were originally created by the [last WAVES participant][blogPost] that worked with Anya, which were then built upon and edited for this summer. 
+
+  The constructors for each organism were edited to be up-to-date, which additional functions to add symbionts, which would then increase the count when simulating. 
+
+  For creating the world, a similar approach was taken for simulating a world in Symbulation. Since we didn't need all the functions and other attributes as the SymWorld in Symbulation, we inherited certain functions that were necessary for this project. For our world, an empirical world constructor was used to birth the organisms. In the world, two different selection methods were used; elite select and tournament select. Elite select picks a set of the most fit individuals for the next generation while tournament select randomly samples a subset of the population and then chooses the most-fit individual in that sample to progress to the next generation. These functions were originally created in Empirical, but changed slightly to fit this project, some of which included things such as passing the pointer to DoBirth. The DoBirth function is where I spent a fair amount of time with, which is shown for tournament select below.
```

Comment:
What does the pointer represent and why is it passed? If you can you highlight it in code or describe it that would help with the context.